resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -119,6 +126,12 @@ class ExecutorPodsAllocator(
       }
     }
     snapshotsStore.addSubscriber(podAllocationDelay) {
+      maxNumExecutorFailuresOpt.foreach { maxNumExecutorFailures =>
+        if (failureTracker.numFailedExecutors > maxNumExecutorFailures) {
+          logError(s"Max number of executor failures ($maxNumExecutorFailures) reached")
+          SparkContext.getActive.foreach(_.stop(EXIT_MAX_EXECUTOR_FAILURES))
```

Comment:
It's ugly, but `AbstractPodsAllocator` is a developer api, pass `sc` will change the constructor signature. Suggestions are welcome.